client/app/bundles/comments/startup/ServerApp.jsx

```diff
@@ -1,13 +1,13 @@
 import React from 'react';
 import { Provider } from 'react-redux';
-
-import createStore from '../store/commentsStore';
 import NonRouterCommentsContainer from '../containers/NonRouterCommentsContainer';
+import NavigationBarContainer from '../containers/NavigationBarContainer';
 
-export default props => {
-  const store = createStore(props);
+export default (_props, _railsContext) => {
+  const store = ReactOnRails.getStore('commentsStore');
   return (
     <Provider store={store}>
+      <NavigationBarContainer />
```

Comment:
@justin808 Here's your problem - there are two components inside of the provider block.